frontend/kesaseteli/youth/src/components/header/__tests__/Header.test.tsx

```diff
@@ -0,0 +1,21 @@
+import Header from 'kesaseteli/youth/components/header/Header';
+import headerApi from 'kesaseteli-shared/__tests__/utils/component-apis/header-api';
+import renderComponent from 'kesaseteli-shared/__tests__/utils/components/render-component';
+import React from 'react';
+import { waitFor } from 'shared/__tests__/utils/test-utils';
+import { SUPPORTED_LANGUAGES } from 'shared/i18n/i18n';
+
+describe('frontend/kesaseteli/youth/src/components/header/Header.tsx', () => {
+  it('can change supported language', async () => {
```

Comment:
I think this is a good test case for using [expect.assertions(x)](https://jestjs.io/docs/expect#expectassertionsnumber)

Reply:
good to know! I added this

Reply:
The idea is good byt it does not work:
```
 Expected three assertions to be called but received six assertion calls.
@frontend/employer:       46 |     renderComponent(<Header />, { push: spyRouterPush });
@frontend/employer:       47 |     await headerApi.expectations.userIsLoggedOut();
@frontend/employer:     > 48 |     expect.assertions(SUPPORTED_LANGUAGES.length);
@frontend/employer:          |            ^
@frontend/employer:       49 |     for (const lang of SUPPORTED_LANGUAGES) {
@frontend/employer:       50 |       headerApi.actions.changeLanguage('fi', lang);
@frontend/employer:       51 |       await waitFor(() =>

```

the waitFor can be called several times until expectation is true, so there is arbitrary amount of async calls.